apps/src/blockly/addons/cdoFieldFlyout.js

```diff
@@ -87,19 +90,17 @@ export default class CdoFieldFlyout extends GoogleBlockly.Field {
     if (!this.isVisible()) {
       return new Blockly.utils.Size(0, 0);
     }
+    // On first render, if the flyout is not visible and it should be,
+    // show the flyout. We can't show the flyout until other components are
+    // rendered, so we delay showing it until here.
+    if (this.flyoutShouldBeVisible && !this.isFlyoutVisible()) {
+      this.setFlyoutVisible(true);
+    }
     // Normally, Blockly skips rendering fields unless we've notified the
     // rendering system that they've changed (this.isDirty_), but in this
-    // case, we want to always re-render / resize the field so that it
-    // matches the width of the block.
-    this.render_();
-
-    if (this.visible_ && this.size_.width === 0) {
-      // If the field is not visible the width will be 0 as well, one of the
-      // problems with the old system.
-      this.render_();
-    }
-
-    return this.size_;
+    // case, we want to always re-render / resize the field.
+    this.isDirty_ = true;
```

Comment:
Great that this can work this way without having to call `render_()` like we were doing before!